doc/api/util.md

```diff
@@ -847,6 +847,93 @@ added: v8.0.0
 * {symbol} that can be used to declare custom promisified variants of functions,
 see [Custom promisified functions][].
 
+## util.report.getReport([err])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `err` {Object}
+* Returns: {Object} Returns the diagnostics report as an Object
+
+Generates a JSON-formatted diagnostic report summary of the running process.
+The report includes JavaScript and native stack traces, heap statistics,
+platform information, resource usage etc. 
+
+```js
+const { report } = require('util');
+const data = report.getReport();
+console.log(data);
+```
+Additional documentation on diagnostic report is available
+at [node-report documentation][]
+
+## util.report.setDiagnosticReportOptions([options]);
```

Comment:
Question … would we still need this function if we had an API to modify CLI flags after the process has started, similar to `v8.setFlagsFromString()`?

Reply:
1. `setDiagnosticReportOptions` is high level and hence more consumable I guess
2. `v8.setFlagsFromString()` alters only `v8` tunables, does not alter node's own flags IIUC - I don't see any callbacks comes from this function to Node land for Node to refresh its private configuration?

Reply:
@gireeshpunathil Yes, this was a hypothetical question – what if we *had* something that did this for Node’s own flags?

Reply:
ok. :) . In that case, this function can be still a wrapper, that collects the user interest i bulk and individually call that hypothetical API for each flags; further if that API is found to be highly consumable than this function, this will become redundant, along with my `nodereport::SyncConfig` method. :(